types/graphql-upload/index.d.ts

```diff
@@ -46,3 +46,8 @@ export interface FileUpload {
   encoding: string;
   createReadStream(): ReadStream;
 }
+
+export class Upload {
+  resolve(file: FileUpload): void;
```

Comment:
I'm a little surprised this doesn't support a `.then` call, but this signature appears to match the implementation.

Reply:
It technically should _also_ have a `promise` property of type `Promise<FileUpload>`; however this PR was certainly more correct than the status quo (which omitted the type entirely) and should realistically work for most external use-cases (since only the library would need to access that field).